### PR TITLE
XW-3816 | Fix fallback navigation 

### PR DESCRIPTION
**Pull request by hakubo**

### CODE REVIEW

**Review by rybmat — APPROVED** [review]

File: includes/wikia/services/ApiService.class.php [hunk 1]
```diff
@@ -51,14 +51,11 @@ static function call( Array $params ) {
 	 *
 	 * @return mixed API response
 	 */
-	static function foreignCall( $dbName, Array $params, $endpoint = self::API, $setUser = false ) {
-		wfProfileIn(__METHOD__);
-
+	static function foreignCall( string $dbName, Array $params, string $endpoint = self::API, bool $setUser = false ) {
```
Comment (rogatty) [inline includes/wikia/services/ApiService.class.php]:
```
There was 1 failure:

1) ApiAccessServiceTest::testForeignCall
Array (...) does not match expected type "boolean".
```

File: includes/wikia/services/ApiService.class.php [hunk 2]
```diff
@@ -95,18 +90,20 @@ static function foreignCall( $dbName, Array $params, $endpoint = self::API, $set
...
 	 */
-	private static function getHostByDbName( $dbName ) {
+	private static function getHostByDbName( string $dbName ) {
 		global $wgDevelEnvironment, $wgDevDomain;
 
-		$cityId = WikiFactory::DBtoID( $dbName );
-		$hostName = WikiFactory::getVarValueByName( 'wgServer', $cityId );
-
 		if ( !empty( $wgDevelEnvironment ) ) {
+			$hostName = WikiFactory::DBtoUrl( $dbName );
```
Comment (rogatty) [inline includes/wikia/services/ApiService.class.php]: Why do we want to use `DBtoUrl` on devboxes and `wgServer` on the rest of envs? If there is a good reason for that, please add a comment to the code.

File: includes/wikia/models/DesignSystemCommunityHeaderModel.class.php [hunk 3]
```diff
@@ -112,23 +114,41 @@ public function getNavigation(): array {
...
 
-			$this->wikiLocalNavigation = $this->formatLocalNavData(
-				( new NavigationModel() )->getFormattedWiki(
-					NavigationModel::WIKI_LOCAL_MESSAGE,
-					$wikitext
-				)['wiki'],
-				1
-			);
+			//No need for foreignCall if we're getting data for the same wiki
+			if ($this->productInstanceId == $wgCityId) {
```
Comment (rogatty) [inline includes/wikia/models/DesignSystemCommunityHeaderModel.class.php]: A lot of missing spaces in this file, please reformat.

**Review by rogatty — CHANGES_REQUESTED** [review]
Please fix the tests, explain the changes in `getHostByDbName` and it will LGTM.

**Review by rogatty — APPROVED** [review]

**Pull request merged**